src/qcodes_contrib_drivers/drivers/Thorlabs/Thorlabs_PM100D/tlpm.py

```diff
@@ -0,0 +1,146 @@
+        dll_path = pathlib.Path(vxi_64, 'Win64', 'Bin', 'TLPM_64.dll')
+    else:
+        dll_path = pathlib.Path(vxi_32, 'WinNT', 'Bin', 'TLPM_32.dll')
+
+    os.add_dll_directory(str(dll_path.parent))
+    sys.path.append(str(pathlib.Path(vxi_32, 'WinNT', 'TLPM', 'Examples',
+                                     'Python')))
+
+    import TLPM
+
+
+class ThorlabsPM100D(Instrument):
+    """TLPM driver for Thorlabs PM100D power meter.
+
+    This driver wraps the example Python driver bundled with the
+    Optical Power Monitor software which can be downloaded `here`_.
+
+    .. _here: https://www.thorlabs.com/software_pages/ViewSoftwarePage.cfm?Code=OPM
+
+    Args:
+        name:
+            An identifier for this instrument.
+        address (optional):
+            The USB address for the device. If omitted, the first
+            available device found will be used.
+        reset (optional):
+            Reset the instrument on connection. Defaults to False.
+        thorlabs_tlpm (optional):
+            An instance of the :class:`TLPM` class from the power
+            monitor software examples.
+        metadata (optional):
+            Additional static metadata.
+        label (optional):
+            Nicely formatted name of the instrument.
+
+    Raises:
+        FileNotFoundError:
+            If the dll is not found.
+
+    """
+
+    def __init__(self, name: str, address: str = '', reset: bool = False,
+                 thorlabs_tlpm: TLPM.TLPM | None = None,
+                 metadata: Mapping[Any, Any] | None = None,
+                 label: str | None = None):
+        if sys.platform != 'win32':
+            raise NotImplementedError('This driver is only available on '
+                                      'Windows.')
+        if vxi_32 is None:
+            raise FileNotFoundError('IVI VXIPNP path not detected.')
+
+        self.tlpm = thorlabs_tlpm or TLPM.TLPM()
+        # NEED to call with IDQuery==True, otherwise the following error is
+        # raised: NameError: The given session or object reference does not
+        # support this operation.
+        self.tlpm.open(address.encode() or self._search_for_device(), True,
+                       reset)
+
+        super().__init__(name, metadata, label)
+
+        self.add_parameter('power',
+                           get_cmd=self._get_power,
+                           label='Power',
+                           unit='W')
+        self.add_parameter('wavelength',
+                           get_cmd=self._get_wavelength,
+                           set_cmd=self._set_wavelength,
+                           label='Wavelength',
+                           unit='nm')
+        self.add_parameter('averaging_time',
+                           get_cmd=self._get_averaging_time,
+                           set_cmd=self._set_averaging_time,
+                           label='Averaging time',
+                           unit='s')
```

Comment:
```suggestion
        self.power = Parameter('power',
                           get_cmd=self._get_power,
                           label='Power',
                           unit='W')
        self.wavelength = Parameter('wavelength',
                           get_cmd=self._get_wavelength,
                           set_cmd=self._set_wavelength,
                           label='Wavelength',
                           unit='nm')
        self.averaging_time = Parameter('averaging_time',
                           get_cmd=self._get_averaging_time,
                           set_cmd=self._set_averaging_time,
                           label='Averaging time',
                           unit='s')
```
It's better to define the attributes statically for documentation purpose.